src/components/Admin/UserPage/DeleteOldDataModal.tsx

```diff
@@ -0,0 +1,147 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  CheckboxGroup,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  useDisclosure,
+  Text,
+  Divider,
+  Flex,
+} from '@chakra-ui/react';
+import React, {useState} from 'react';
+import {useTranslation} from 'react-i18next';
+import {FaTrash} from 'react-icons/fa';
+import {useAuth} from '../../../contexts/AuthContext';
+import {firestore} from '../../../firebase';
+
+function DeleteOldDataModal(): JSX.Element {
+  // --------------- State maintenance variables ------------------------
+  const {isOpen, onOpen, onClose} = useDisclosure();
+  const [areDisclosuresChecked, setAreDisclosuresChecked] = useState([
+    false,
+    false,
+    false,
+  ]);
+
+  const allDisclosuresChecked = areDisclosuresChecked.every(Boolean);
+  const {isAdmin} = useAuth();
+
+  const {t} = useTranslation('administration');
+
+  /**
+   * Resets modal state values before closing the modal.
+   */
+  function handleClose(): void {
+    // Modal state
+    setAreDisclosuresChecked([false, false, false]);
+    onClose();
+  }
+
+  /**
+   * Marks data older than a defined time as ready for deletion by a Cloud Function
+   */
+  async function markOldDataForDeletion(): Promise<void> {
+    if (allDisclosuresChecked && isAdmin) {
+      const sensorsList = await firestore.collection('sensors').get();
+      const deletionDocRef = firestore.collection('deletion').doc('todo');
+
+      // Get current mapping or set to empty
+      const mapping =
+        (await (await deletionDocRef.get())?.data()?.data) ??
+        Object.create(null);
+
+      const offset = 7;
+      const relevantTime = new Date();
+      relevantTime.setDate(relevantTime.getDate() - offset);
+
+      for (const sensorDoc of sensorsList.docs) {
+        mapping[sensorDoc.id] = relevantTime;
+      }
+
+      deletionDocRef.update({data: mapping});
+    }
+  }
+
+  function handleSubmit(): void {
+    markOldDataForDeletion().then(handleClose);
+  }
+
+  return (
+    <Box marginY={2}>
+      <Button colorScheme="red" onClick={onOpen}>
+        {t('deleteOldData.launchButton')}
+      </Button>
+      <Modal isOpen={isOpen} onClose={handleClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>{t('deleteOldData.modalHeader')}</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <Text>{t('deleteOldData.body.paragraph1')}</Text>{' '}
+            <Text>{t('deleteOldData.body.paragraph2')}</Text>
+            <Divider marginTop={2} marginBottom={2} />
+            <CheckboxGroup>
+              <Stack mt={1} spacing={1}>
+                <Checkbox
+                  isChecked={areDisclosuresChecked[0]}
+                  onChange={e =>
+                    setAreDisclosuresChecked([
+                      e.target.checked,
```

Comment:
maybe just use separate state for each checkbox?

Reply:
This was what [the ChakraUI documentation](https://chakra-ui.com/docs/form/checkbox#indeterminate) suggested. Should I seperate instead? I think it was because we want to be able to use the boolean.all

Reply:
Perhaps have a helper function that says which index to update, the code just seems repetitive/error prone. boolena.all is nice, but it might be worth doing `cond1 && cond2 && cond3` for readability